cli/Sources/TuistGenerator/Generator/ProjectDescriptorGenerator.swift

```diff
@@ -306,6 +311,31 @@ final class ProjectDescriptorGenerator: ProjectDescriptorGenerating {
         pbxProject.localPackages = localPackageReferences.sorted { $0.key < $1.key }.map { $1 }
     }
 
+    private func addLocalPackageReference(
+        root: PBXGroup,
+        groupPath: String,
+        reference: PBXFileReference
+    ) throws {
+        let packagesGroup: PBXGroup?
+        if let existing = root.group(named: "Packages") {
```

Comment:
What about fetching or creating this one earlier, such that we don't do the look up in every call to `addLocalPackageReference`?

Reply:
I'd expect the signature to be:

```swift
private func addLocalPackageReference(
        root: PBXGroup,
        groupPath: String,
        reference: PBXFileReference,
        packagesGroup: PBXGroup
    )
```